packages/components/src/composite/index.ts

```diff
@@ -1,7 +1,137 @@
-// Originally this pointed at a Reakit implementation of
-// `Composite`, but we are removing Reakit entirely from the
-// codebase. We will continue to support the Reakit API
-// through the 'legacy' version, which uses Ariakit under
-// the hood.
+/**
+ * Composite is a component that may contain navigable items represented by
+ * Composite.Item. It's inspired by the WAI-ARIA Composite Role and implements
+ * all the keyboard navigation mechanisms to ensure that there's only one
+ * tab stop for the whole Composite element. This means that it can behave as
+ * a roving tabindex or aria-activedescendant container.
+ *
+ * @see https://ariakit.org/components/composite
+ */
 
-export * from './legacy';
+/**
+ * External dependencies
+ */
+import * as Ariakit from '@ariakit/react';
+import type { CompositeStore, CompositeStoreProps } from '@ariakit/react';
+
+/**
+ * Creates a composite store.
+ * @see https://ariakit.org/reference/use-composite-store
+ * @example
+ * ```jsx
+ * const useCompositeStore = useCompositeStore();
+ * const store = useCompositeStore();
+ * <Composite store={store}>
+ *   <Composite.Item>Item</Composite.Item>
+ *   <Composite.Item>Item</Composite.Item>
+ *   <Composite.Item>Item</Composite.Item>
+ * </Composite>
+ * ```
+ */
+export const useCompositeStore = Ariakit.useCompositeStore;
+
+const Group = Ariakit.CompositeGroup;
+Group.displayName = 'Composite.Group';
+
+export const GroupLabel = Ariakit.CompositeGroupLabel;
+GroupLabel.displayName = 'Composite.GroupLabel';
+
+export const Item = Ariakit.CompositeItem;
+Item.displayName = 'Composite.Item';
+
+export const Row = Ariakit.CompositeRow;
+Row.displayName = 'Composite.Row';
+
+/**
+ * Renders a composite widget.
+ * @see https://ariakit.org/reference/composite
+ * @example
+ * ```jsx
+ * const useCompositeStore = useCompositeStore();
+ * const store = useCompositeStore();
+ * <Composite store={store}>
+ *   <Composite.Item>Item 1</Composite.Item>
+ *   <Composite.Item>Item 2</Composite.Item>
+ * </Composite>
+ * ```
+ */
+export const Composite = Object.assign( Ariakit.Composite, {
+	/**
```

Comment:
Just me or is the display name rendering as `Composite2`? Not sure what that's about, but I think we may need to add an explicit `displayName` here.

```suggestion
export const Composite = Object.assign( Ariakit.Composite, {
	displayName: 'Composite',
	/**
```

Reply:
Interesting. I think this was potentially a side-effect of the ariakit version update? After rebasing, I started getting errors when setting the `displayName` on ariakit components. I had to resort to `Object.assign` for the subcomponents too.

https://github.com/WordPress/gutenberg/pull/63564/commits/30d290e72c4f96de3dc2a4cb971c0d8bbcb12992

I will reflect this new best practice in #63714 too.